tests/serverpod_test_server/test/service_protocol_test.dart

```diff
@@ -296,6 +296,259 @@ void main() {
+
+        expect(table.indexes, hasLength(2));
+        expect(table.indexes[0].indexName, 'object_with_index_pkey');
+        expect(table.indexes[0].isPrimary, true);
+        expect(table.indexes[0].isUnique, true);
+        expect(table.indexes[0].predicate, isNull);
+        expect(table.indexes[0].tableSpace, isNull);
+        expect(table.indexes[0].type, 'btree');
+        expect(table.indexes[0].elements, hasLength(1));
+        expect(table.indexes[0].elements.first.type,
+            service.IndexElementDefinitionType.column);
+        expect(table.indexes[0].elements.first.definition, 'id');
+
+        expect(table.indexes[1].indexName, 'object_with_index_test_index');
+        expect(table.indexes[1].isPrimary, false);
+        expect(table.indexes[1].isUnique, false);
+        expect(table.indexes[1].predicate, isNull);
+        expect(table.indexes[1].tableSpace, isNull);
+        expect(table.indexes[1].type, 'brin');
+        expect(table.indexes[1].elements, hasLength(2));
+        expect(table.indexes[1].elements[0].type,
+            service.IndexElementDefinitionType.column);
+        expect(table.indexes[1].elements[0].definition, 'indexed');
+        expect(table.indexes[1].elements[1].type,
+            service.IndexElementDefinitionType.column);
+        expect(table.indexes[1].elements[1].definition, 'indexed2');
+      });
+
+      test('validate dart types', () async {
+        var definition =
+            await serviceClient.insights.getDesiredDatabaseDefinition();
+
+        var columns = definition.tables
+            .firstWhere((table) => table.name == 'object_with_object')
+            .columns
+            .map((c) => c.dartType)
+            .toList();
+
+        expect(columns, hasLength(7));
+        expect(
+            columns,
+            containsAll([
+              'int?',
+              'protocol:SimpleData',
+              'protocol:SimpleData?',
+              'List<protocol:SimpleData>',
+              'List<protocol:SimpleData>?',
+              'List<protocol:SimpleData?>',
+              'List<protocol:SimpleData?>?',
+            ]));
+
+        var columnsWithScopes = definition.tables
+            .firstWhere((table) => table.name == 'object_field_scopes')
+            .columns
+            .map((c) => c.dartType)
+            .toList();
+
+        expect(columnsWithScopes, hasLength(3));
+        expect(columnsWithScopes, containsAll(['int?', 'String', 'String?']));
+      });
+    });
+
+    group('current definition', () {
+      test('matches desired', () async {
+        var desired =
+            await serviceClient.insights.getDesiredDatabaseDefinition();
+        var current =
+            await serviceClient.insights.getCurrentDatabaseDefinition();
+
+        current.matchesDesired(desired);
+      });
+    });
+  });
+}
+
+extension on service.DatabaseDefinition {
+  void matchesDesired(service.DatabaseDefinition desired) {
```

Comment:
`matchesDefinition`?

Reply:
Better? dc0acb76

Due to `if (definition.managed ?? false) {` `target.matchesDefinition(actual) == actual.matchesDefinition(target)` is almost true.

@vlidholt Please resolve if it's fine now.